transports/quic/src/upgrade.rs

```diff
@@ -70,8 +70,7 @@ impl Future for Connecting {
             };
             match event {
                 quinn_proto::Event::Connected => {
-                    let session = connection.crypto_session();
-                    let identity = session
+                    let identity = connection
                         .peer_identity()
                         .expect("connection got identity because it passed TLS handshake; qed");
```

Comment:
Is this guaranteed by how we construct `Connection`? If so, we could panic inside `peer_identity`.

Reply:
It's only guaranteed once the handshake succeeded, thus only safe to panic here because we received the `Event::Connected`.